cabal-install/src/Distribution/Client/Init/NonInteractive/Command.hs

```diff
@@ -274,14 +274,22 @@ licenseHeuristics :: Interactive m => InitFlags -> m SpecLicense
 licenseHeuristics flags = getLicense flags $ guessLicense flags
 
 -- | The author's name. Prompt, or try to guess from an existing
---   darcs repo.
+--   git repo.
 authorHeuristics :: Interactive m => InitFlags -> m String
-authorHeuristics flags = getAuthor flags guessAuthorEmail
+authorHeuristics flags = do
+  email' <- guessAuthorEmail
```

Comment:
same as above, could be 
```haskell
authorHeuristics flags = guessAuthorEmail >>=
  maybe (getAuthor flags $ return "") (getAuthor flags . return)
```

By the way, you ought to fix the name of the binding here: `email'` does not belong here.

Reply:
Yeah this is what I mentioned before: in non-interactive the functions seem to be switched, but this was already the case before I touched it so it seems to work..

Reply:
@BasLaa oh, I missed that comment of yours, sorry! I just checked that it's complete bonkers! Could you, please, you fix it (call `guessAuthor` from `authorHeuristics` and similarly for email)?

Reply:
That's what I thought as well :)